config.go

```diff
@@ -66,16 +68,55 @@ func LoadConfigurationFile(path string) (Config, error) {
 	return conf, err
 }
 
+func GetOSConfigPath(homePath string) string {
+	var configDir string
+
+	defaultConfig := map[string]string{
+		"windows": "LocalAppData",
+		"darwin":  "Library/Application Support",
+		"linux":   ".config",
+	}
+
+	goos := runtime.GOOS
+	switch goos {
+	case "windows":
+		configDir = os.Getenv("LOCALAPPDATA")
+	case "darwin", "linux":
+		configDir = os.Getenv("XDG_CONFIG_HOME")
+	case "default":
+		// exit with no path for unknown os
+		return ""
```

Comment:
Let's use `XDG_CONFIG_HOME` for all non-windows OSes, and default to `~/.config` for everything except Windows/Darwin.

Reply:
Like this?

```diff
-       case "default":
-               // exit with no path for unknown os
-               return ""
+       default:
+               configDir = filepath.Join(homePath, ".config")
        }
```

Reply:
Ideally we would use `XDG_CONFIG_HOME` if it exists and otherwise fall back to `~/.config`. Actually let's also fall back to `~/.config` for darwin. So you could do something like 

```go
var goos string
switch (runtime.GOOS) {
case "windows":
    configDir = os.Getenv("LOCALAPPDATA")
    goos = "windows"
default:
    configDir = os.Getenv("XDG_CONFIG_HOME")
    goos = "default"
}
```
and you can also remove the `"darwin"` entry in `defaultConfig`, and change the linux one to `"default"`